src/services/completions.ts

```diff
@@ -675,6 +675,37 @@ namespace ts.Completions {
             hasAction = !importCompletionNode;
         }
 
+        const kind = SymbolDisplay.getSymbolKind(typeChecker, symbol, location); // TODO: GH#18217
+        if (kind === ScriptElementKind.jsxAttribute && preferences.includeCompletionsWithSnippetText && preferences.jsxAttributeCompletionStyle && preferences.jsxAttributeCompletionStyle !== "none") {
+            let useBraces = preferences.jsxAttributeCompletionStyle === "braces";
+            const type = typeChecker.getTypeOfSymbolAtLocation(symbol, location);
+
+            // If is boolean like or undefined, don't return a snippet we want just to return the completion.
+            if (preferences.jsxAttributeCompletionStyle === "auto"
+                && !(type.flags & TypeFlags.BooleanLike)
+                && !(type.flags & TypeFlags.Union && find((type as UnionType).types, type => !!(type.flags & TypeFlags.BooleanLike)))
+            ) {
+                if (type.flags & TypeFlags.StringLike || (type.flags & TypeFlags.Union && every((type as UnionType).types, type => !!(type.flags & (TypeFlags.StringLike | TypeFlags.Undefined))))) {
+                    // If is string like or undefined use quotes
+                    insertText = `${name}=${quote(sourceFile, preferences, "$1")}`;

```

Comment:
Oh, I almost forgot—I’m pretty sure my import statement completions code has this same bug, but @mjbvz reminded me that we need to escape `$` in the completed text whenever we set `isSnippet` to `true`. Test:

```ts
interface Props {
  this$1IsAValid$2PropName: string;
}

declare function Component(props: Props): any;

(<Component th/**/ />);
```

Unfortunately fourslash isn’t really going to be able to catch this mistake through anything other than manual assertion that the `insertText` contains a backslash before `$`.

Reply:
Good catch, fixed. I have also fixed the completions for imports.